samples/subsys/usb/webusb/src/msosv2.h

```diff
@@ -0,0 +1,116 @@
+	'E', 0x00, '-', 0x00, '9', 0x00, 'D', 0x00, '8', 0x00, '2', 0x00, \
+	'-', 0x00, '4', 0x00, '9', 0x00, '1', 0x00, '9', 0x00, '6', 0x00, \
+	'3', 0x00, '9', 0x00, '5', 0x00, '6', 0x00, '2', 0x00, 'B', 0x00, \
+	'3', 0x00, '}', 0x00, 0x00, 0x00, 0x00, 0x00
+
+struct msosv2_descriptor {
+	struct msosv2_descriptor_set_header header;
+	struct msosv2_compatible_id compatible_id;
+	struct msosv2_guids_property guids_property;
+} __packed;
+
+static struct msosv2_descriptor msosv2_desc = {
+	.header = {
+		.wLength = sizeof(struct msosv2_descriptor_set_header),
+		.wDescriptorType = MS_OS_20_SET_HEADER_DESCRIPTOR,
+		.dwWindowsVersion = sys_cpu_to_le32(SAMPLE_MSOS2_OS_VERSION),
+		.wTotalLength = sizeof(msosv2_desc),
+	},
+	.compatible_id = {
+		.wLength = sizeof(struct msosv2_compatible_id),
+		.wDescriptorType = MS_OS_20_FEATURE_COMPATIBLE_ID,
+		.CompatibleID = {'W', 'I', 'N', 'U', 'S', 'B', 0x00, 0x00},
+	},
+	.guids_property = {
+		.wLength = sizeof(struct msosv2_guids_property),
+		.wDescriptorType = MS_OS_20_FEATURE_REG_PROPERTY,
+		.wPropertyDataType = MS_OS_20_PROPERTY_DATA_REG_MULTI_SZ,
+		.wPropertyNameLength = 42,
+		.PropertyName = {DEVICE_INTERFACE_GUIDS_PROPERTY_NAME},
+		.wPropertyDataLength = 80,
+		.bPropertyData = {WEBUSB_DEVICE_INTERFACE_GUID},
+	},
+};
+
+struct bos_msosv2_descriptor {
+	struct usb_bos_platform_descriptor platform;
+	struct usb_bos_capability_msos cap;
+} __packed;
+
+struct bos_msosv2_descriptor bos_msosv2_desc = {
+	/*
+	 * Microsoft OS 2.0 Platform Capability Descriptor,
+	 * see Microsoft OS 2.0 Descriptors Specification
+	 */
+	.platform = {
+		.bLength = sizeof(struct usb_bos_platform_descriptor)
+			 + sizeof(struct usb_bos_capability_msos),
+		.bDescriptorType = USB_DESC_DEVICE_CAPABILITY,
+		.bDevCapabilityType = USB_BOS_CAPABILITY_PLATFORM,
+		.bReserved = 0,
+		/* Microsoft OS 2.0 descriptor platform capability UUID
+		 * D8DD60DF-4589-4CC7-9CD2-659D9E648A9F
+		 */
+		.PlatformCapabilityUUID = {
+			0xDF, 0x60, 0xDD, 0xD8,
+			0x89, 0x45,
+			0xC7, 0x4C,
+			0x9C, 0xD2,
+			0x65, 0x9D, 0x9E, 0x64, 0x8A, 0x9F,
+		},
+	},
+	.cap = {
+		.dwWindowsVersion = sys_cpu_to_le32(SAMPLE_MSOS2_OS_VERSION),
+		.wMSOSDescriptorSetTotalLength = sys_cpu_to_le16(sizeof(msosv2_desc)),
+		.bMS_VendorCode = SAMPLE_MSOS2_VENDOR_CODE,
+		.bAltEnumCode = 0x00
+	},
+};
+
+static int msosv2_to_host_cb(const struct usbd_context *const ctx,
+			     const struct usb_setup_packet *const setup,
+			     struct net_buf *const buf)
+{
+	LOG_INF("Vendor callback to host");
+
+	if (setup->bRequest == SAMPLE_MSOS2_VENDOR_CODE &&
+	    setup->wIndex == MS_OS_20_DESCRIPTOR_INDEX) {
+		LOG_INF("Get MS OS 2.0 Descriptor Set");
+
+		net_buf_add_mem(buf, &msosv2_desc,
+				MIN(net_buf_tailroom(buf), sizeof(msosv2_desc)));
+
+		return 0;
+	}
+
+	return -ENOTSUP;
+}
+
+USBD_DESC_BOS_VREQ_DEFINE(bos_vreq_msosv2, sizeof(bos_msosv2_desc), &bos_msosv2_desc,
+			  SAMPLE_MSOS2_VENDOR_CODE, msosv2_to_host_cb, NULL);
```

Comment:
Why this is defined in a header?

Reply:
Why not? Or what is the problem? This is limited to a specific part and can be used as a template.

Reply:
Everything compilation unit that includes this header will have the variables in the corresponding object file.

Reply:
There is only that include it. So where is the problem?